ibmjava/update-multiarch.sh

```diff
@@ -0,0 +1,409 @@
+print_ubuntu_os() {
+	cat >> $1 <<-EOI
+	FROM ubuntu:16.04
+
+	EOI
+}
+
+# Print the supported Alpine OS
+print_alpine_os() {
+	cat >> $1 <<-EOI
+	FROM alpine:3.6
+
+	EOI
+}
+
+# Print the maintainer
+print_maint() {
+	cat >> $1 <<-EOI
+	MAINTAINER Dinakar Guniguntala <dinakar.g@in.ibm.com> (@dinogun)
+	EOI
+}
+
+# Select the ubuntu OS packages
+print_ubuntu_pkg() {
+	cat >> $1 <<'EOI'
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends wget ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+EOI
+}
+
+# Select the alpine OS packages.
+# Install GNU glibc as J9 needs it, install libgcc_s.so from gcc-libs.tar.xz (archlinux)
+print_alpine_pkg() {
+	cat >> $1 <<'EOI'
+
+RUN apk --update add --no-cache ca-certificates curl openssl xz \
+    && GLIBC_VER="2.25-r0" \
+    && ALPINE_GLIBC_REPO="https://github.com/sgerrand/alpine-pkg-glibc/releases/download" \
+    && curl -Ls ${ALPINE_GLIBC_REPO}/${GLIBC_VER}/glibc-${GLIBC_VER}.apk > /tmp/${GLIBC_VER}.apk \
+    && apk add --allow-untrusted /tmp/${GLIBC_VER}.apk \
+    && curl -Ls https://www.archlinux.org/packages/core/x86_64/gcc-libs/download > /tmp/gcc-libs.tar.xz \
+    && mkdir /tmp/gcc \
+    && tar -xf /tmp/gcc-libs.tar.xz -C /tmp/gcc \
+EOI
+
+	if [ "${ver}" == "9" ]; then
+		GLIBC_PKGS="&& mv /tmp/gcc/usr/lib/libgcc* /tmp/gcc/usr/lib/libstdc++* /usr/glibc-compat/lib \\"
+	else
+		GLIBC_PKGS="&& mv /tmp/gcc/usr/lib/libgcc* /usr/glibc-compat/lib \\"
+	fi
+
+	cat >> $1 <<-EOI
+    ${GLIBC_PKGS}
+    && apk del curl \\
+    && rm -rf /tmp/\${GLIBC_VER}.apk /tmp/gcc /tmp/gcc-libs.tar.xz /var/cache/apk/*
+EOI
+}
+
+# Print the Java version that is being installed here
+print_env() {
+	srcpkg=$2
+	shasums="${srcpkg}"_"${ver}"_sums
+	jverinfo=${shasums}[version]
+	eval jver=\${$jverinfo}
+
+	cat >> $1 <<-EOI
+
+ENV JAVA_VERSION ${jver}
+
+EOI
+}
+
+# OS independent portion (Works for both Alpine and Ubuntu)
+# For Java 9 we use jlink to derive the JRE and the SFJ images.
+print_java_install() {
+	cat >> $1 <<-EOI
+       amd64|x86_64) \\
+         ESUM='$(sarray=${shasums}[x86_64]; eval esum=\${$sarray}; echo ${esum})'; \\
+         YML_FILE='${srcpkg}/linux/x86_64/index.yml'; \\
+         ;; \\
+       i386) \\
+         ESUM='$(sarray=${shasums}[i386]; eval esum=\${$sarray}; echo ${esum})'; \\
+         YML_FILE='${srcpkg}/linux/i386/index.yml'; \\
+         ;; \\
+       ppc64el|ppc64le) \\
+         ESUM='$(sarray=${shasums}[ppc64le]; eval esum=\${$sarray}; echo ${esum})'; \\
+         YML_FILE='${srcpkg}/linux/ppc64le/index.yml'; \\
+         ;; \\
+       s390) \\
+         ESUM='$(sarray=${shasums}[s390]; eval esum=\${$sarray}; echo ${esum})'; \\
+         YML_FILE='${srcpkg}/linux/s390/index.yml'; \\
+         ;; \\
+       s390x) \\
+         ESUM='$(sarray=${shasums}[s390x]; eval esum=\${$sarray}; echo ${esum})'; \\
+         YML_FILE='${srcpkg}/linux/s390x/index.yml'; \\
+         ;; \\
+       *) \\
+         echo "Unsupported arch: ${ARCH}"; \\
```

Comment:
Since this is in an interpolated heredoc, this `${ARCH}` needs to be escaped (comes out as simply `echo "Unsupported arch: "; \`).

Reply:
Thanks for catching that ! Fixed it now.